ratatui-widgets/src/block.rs

```diff
@@ -933,41 +933,56 @@ impl Block<'_> {
     }
 
     /// Render titles in the center of the block
-    ///
-    /// Currently this method aligns the titles to the left inside a centered area. This is not
-    /// ideal and should be fixed in the future to align the titles to the center of the block and
-    /// truncate both sides of the titles if the block is too small to fit all titles.
     #[expect(clippy::similar_names)]
     fn render_center_titles(&self, position: TitlePosition, area: Rect, buf: &mut Buffer) {
+        let mut area = self.titles_area(area, position);
         let titles = self
             .filtered_titles(position, Alignment::Center)
             .collect_vec();
+        // titles are rendered with a space after each title except the last one
         let total_width = titles
             .iter()
-            .map(|title| title.width() as u16 + 1) // space between titles
+            .map(|title| title.width() as u16 + 1)
             .sum::<u16>()
-            .saturating_sub(1); // no space for the last title
-
-        let titles_area = self.titles_area(area, position);
-        let mut titles_area = Rect {
-            x: titles_area.left() + (titles_area.width.saturating_sub(total_width) / 2),
-            ..titles_area
-        };
-        for title in titles {
-            if titles_area.is_empty() {
-                break;
+            .saturating_sub(1);
+
+        if total_width <= area.width {
```

Comment:
For reducing the size of this method, either:
- chop the inner part of the if blocks (e.g. render_center_titles_fitting, render_center_titles_exceeds); or
- see if the for loop can be factored so that the fitting and exceeds parts have a common implementation - these are fairly close with this new approach. I think probably extracting the methods is probably better as it makes it clear that the truncation code isn't necessary unless the titles are chopped.